### PR TITLE
HORIZONAPP-229/iOS3DNonceRequest

### PR DESCRIPTION
**Pull request by ScruffyHipster**
[JIRA Status](https://netsells.atlassian.net/browse/HORIZONAPP-229)

Added method to allow call to braintree to carry out 3D secure check.

### CODE REVIEW

File: ios/Classes/SwiftFlutterBraintreePlugin.swift [hunk 1]
```diff
@@ -85,11 +87,61 @@ public class SwiftFlutterBraintreePlugin: NSObject, FlutterPlugin {
...
+                let viewController: UIViewController = storyboard!.instantiateViewController(withIdentifier: "braintreePaymenetFlowViewController")
+                let navController = UINavigationController(rootViewController: viewController)
+                window.rootViewController = navController
+                navController.pushViewController(self.btPaymentViewController, animated: true)
+            }
+
+            let authToken = string(for: "authorization", in: call)!
+            let amount =  Double(string(for: "amount", in: call)!)
+            let email =  string(for: "email", in: call)!
+            let nonce =  string(for: "none", in: call)!
```
Comment (ptrbrynt) [inline ios/Classes/SwiftFlutterBraintreePlugin.swift]:
```suggestion
            let nonce =  string(for: "nonce", in: call)!
```

**Review by ptrbrynt — CHANGES_REQUESTED** [review]
Looks good other than the typo! Once fixed I'll test it against the Flutter app before merging.

File: ios/Classes/SwiftFlutterBraintreePlugin.swift (hunk 1 above)
Comment (ScruffyHipster) [reply ios/Classes/SwiftFlutterBraintreePlugin.swift]: 🤦‍♂️

Comment (ScruffyHipster) [reply ios/Classes/SwiftFlutterBraintreePlugin.swift]: updated

**Review by ptrbrynt — CHANGES_REQUESTED** [review]
Tested in the app - unfortunately it's crashing at the point of invoking the method.

Here's the stacktrace:

```
2020-03-18 14:46:40.586631+0000 Runner[29770:8680145] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Storyboard (<UIStoryboard: 0x283e2f4e0>) doesn't contain a view controller with identifier 'braintreePaymenetFlowViewController''
*** First throw call stack:
(0x1bb75096c 0x1bb469028 0x1bf9adde4 0x104080670 0x104082c00 0x10228328c 0x10221c5b4 0x1022741e8 0x10222bcf4 0x10222e000 0x1bb6cc77c 0x1bb6cc4b4 0x1bb6cbb28 0x1bb6c69d0 0x1bb6c5f40 0x1c5956534 0x1bf851580 0x100132530 0x1bb544e18)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Pull request closed without merging**